shuffle/cli/src/deploy.rs

```diff
@@ -53,65 +38,165 @@ fn publish_packages_as_transaction(
         derived_address.to_hex_literal()
     );
 
-    // Send a module transaction
-    let seq_number = client
-        .get_account(derived_address)?
-        .into_inner()
-        .ok_or_else(|| anyhow::anyhow!("missing AccountView"))?
-        .sequence_number;
-    let mut new_account = LocalAccount::new(derived_address, new_account_key, seq_number);
-    send_module_transaction(&compiled_package, &client, &mut new_account, &factory)?;
-    check_module_exists(&client, &new_account)
-}
+    let account_seq_number = client.get_account_sequence_number(derived_address).await?;
+    let mut new_account = LocalAccount::new(derived_address, new_account_key, account_seq_number);
 
-pub fn send_module_transaction(
-    compiled_package: &CompiledPackage,
-    client: &BlockingClient,
-    account: &mut LocalAccount,
-    factory: &TransactionFactory,
-) -> Result<()> {
+    let mut all_hashes = Vec::new();
+
+    let compiled_package = build_move_package(project_path.join(MAIN_PKG_PATH).as_ref())?;
     for module in compiled_package
         .transitive_compiled_modules()
         .compute_dependency_graph()
         .compute_topological_order()?
     {
         let module_id = module.self_id();
-        if module_id.address() == &account.address() {
-            println!("Deploying Module: {}", module_id);
-            let mut binary = vec![];
-            module.serialize(&mut binary)?;
-            let publish_txn = account.sign_with_transaction_builder(
-                factory.payload(TransactionPayload::Module(Module::new(binary))),
-            );
+        if module_id.address() != &new_account.address() {
+            println!("Skipping Module: {}", module_id);
+            continue;
+        }
+        println!("Deploying Module: {}", module_id);
+        let mut binary = vec![];
+        module.serialize(&mut binary)?;
+
+        let hash = send_module_transaction(&client, &mut new_account, binary).await?;
+        all_hashes.push(hash);
+    }
+
+    confirm_txns_executed_from_hashes(&client, all_hashes).await
+}
+
+async fn send_module_transaction(
+    client: &DevApiClient,
+    account: &mut LocalAccount,
+    module_binary: Vec<u8>,
+) -> Result<String> {
+    let factory = TransactionFactory::new(ChainId::test());
+    let publish_txn = account.sign_with_transaction_builder(
+        factory.payload(TransactionPayload::Module(Module::new(module_binary))),
+    );
+    let bytes = bcs::to_bytes(&publish_txn)?;
+    let resp = client.post_transactions(bytes).await?;
+    let json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+    let hash = get_hash_from_post_txn(json)?;
+    Ok(hash)
+}
 
-            shared::send_transaction(client, publish_txn)?;
+pub async fn confirm_txns_executed_from_hashes(
```

Comment:
1. Why is this public? Let's remove the `pub`
2. This function is still too dense, let's break it out into more testable methods. Suggestions: pull out what's inside the `for hash in all_hashes`
3. `time::Duration::from_millis(1000)` => `time::Duration::from_second(1)`
There's also a world where you check the hash execution in parallel and join at the end.